### PR TITLE
Check if system buckets exist after the uploader run

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- as in the title

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/11341

### CODE REVIEW

File: internal/bucket/handler.go [hunk 1]
```diff
@@ -102,6 +102,21 @@ func (h *Handler) CreateIfDoesntExist(bucketName, bucketRegion string) error {
...
+// CheckBuckets makes sure that system buckets are created if they don't exist yet
+func (h *Handler) CheckBuckets(buckets SystemBucketNames) error {
+	err := h.CreateIfDoesntExist(buckets.Private, h.cfg.Region)
+	if err != nil {
+		return errors.Wrapf(err, "while creating private system buckets")
+	}
+
+	err = h.CreateIfDoesntExist(buckets.Public, h.cfg.Region)
+	if err != nil {
+		return errors.Wrapf(err, "while creating private system buckets")
```
Comment (tgorgol) [inline internal/bucket/handler.go:114]:
```suggestion
		return errors.Wrapf(err, "while creating public system buckets")
```

File: cmd/uploader/main.go [hunk 2]
```diff
@@ -75,6 +75,10 @@ func main() {
 		err = c.Save(configurer.SharedAppConfig{SystemBuckets: buckets})
 		exitOnError(err, "Error during saving system bucket configuration")
 	} else {
+		handler := bucket.NewHandler(client, cfg.Bucket)
+		err = handler.CheckBuckets(sharedConfig.SystemBuckets)
```
Comment (tgorgol) [inline cmd/uploader/main.go:79]: Take a look at `CreateSystemBuckets` method. It uses `CreateIfDoesntExist` underneath as well but also sets bucket policies which are required. I think you should either add the policy setting logic to your `CheckBuckets` method or refactor `CreateSystemBuckets` to reuse it here.

Comment (pPrecel) [reply cmd/uploader/main.go:79]: Yup. You are right. I figured it out yesterday too

Comment (pPrecel) [reply cmd/uploader/main.go:79]: thanks

**Review by tgorgol — APPROVED** [review]
LGTM

**Pull request merged**